### DISCUSSION TITLE
Toggle to point representation.

### DISCUSSION THREAD

**Discussion by nennigb** [pyvista/pyvista#2128] [Q&A] [answered]
Hi everyone,

I would like to Toggle interactively from wireframe, surface and point representation of data created from `add_mesh`.
I found that 'w' and 's' allow to toggle from wireframe and surface respectively but **how to toggle to point representation** ?
I was thinking that `add_key_event` could be used, but I don't know which properties I must change.

Thanks a lot,

pyvista is great!

**Comment by nennigb** [comment]
After some tests, the following approach works (I don't know if it is the best one):
```python
import pyvista as pv
sphere = pyvista.Sphere()
pl = pv.Plotter()
actor = pl.add_mesh(sphere, render_points_as_spheres=True, point_size=10.0)
prop = actor.GetProperty()
prop.SetRepresentationToPoints()
pl.show()
```
It remains to change the `actor` and update the `Plotter` in the `add_key_event` callback function.

**Comment by banesullivan** [comment, marked as answer]
We used to have this set up to use the `v` key to switch to points representation... not sure why that isn't working.

FWIW, you can add any key bound callback with `add_key_event` https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_key_event.html

```py
import pyvista as pv

mesh = pv.Sphere()

p = pv.Plotter(notebook=0)
actor = p.add_mesh(mesh)

def toggle_points(*args):
    prop = actor.GetProperty()
    if prop.GetRepresentation() == 2:
        prop.SetRepresentationToPoints()
    else:
        prop.SetRepresentationToSurface()
    p.render()

p.add_key_event('v', toggle_points)
p.show()
```

**Reply by banesullivan** [reply]
If you want to do this for all actors, you could iterate over `p.renderer.actors` in the callback

**Reply by nennigb** [reply]
Thanks a lot !
